### ISSUE TITLE
Micronaut rabbitmq does not support concurrent consumption of messages from the same queue

### ISSUE THREAD

**Issue by oscarfh** [micronaut-projects/micronaut-rabbitmq#248]
This issue was discussed in [this gitter conversation](https://gitter.im/micronautfw/questions?at=60e4b5b024f0ae2a2448798f).

## Context
Imagine queue in which the messages represent jobs. Each job is independent from each other. 
There is a Micronaut app that reads these messages and executes these jobs. For the sake of this example, 500ms.

## Problem
[Micronaut's documentation indicates](https://micronaut-projects.github.io/micronaut-rabbitmq/latest/guide/#consumerExecutor) that the consumers run in a thread pool.
The problem is that, while this is true, Micronaut will only register a single consumer to that queue, so, while there is a thread pool ready to consume the messages, because there is only a single consumer, this consumer will only get the next message when the previous message has been ack'ed, therefore the threadpool remains idle, while only a single thread is working.

This can be verified by log messages like this one (this is a manipulated example from my real logs):
```
 2021-07-06 17:18:56.206 BRT consumer-executor-thread-4 "Received message from RabbitMQ.."
 2021-07-06 17:18:57.206 BRT consumer-executor-thread-4 "Received message from RabbitMQ.."
 2021-07-06 17:18:58.206 BRT consumer-executor-thread-4 "Received message from RabbitMQ.."
```
Here we can see that only a single thread is being used to consume the messages, while all others (in this case, 24 other threads) are idle. The 1s different also indicates that the message is being processed sequentially.

## Proposed Solution
Micronaut should register a number of consumers that is equal to the number of threads in the threadpool, in this case, each consumer would  have one available thread to consume its messages.

I am looking at the code to see if I can do this implementation myself.

**Comment by oscarfh** [comment]
I was playing around with the code to see how things were put together.
It seems to be (from my talk with James Kleeh), that we could create multiple channels (one channel per thread), as mentioned.
I had a small change that opened 25 subscribers and I was able to consume messages concurrently.

I was thinking about:
- The ConsumerAdvice should create and initialize one channel per thread in the `consumer` executor
- Ensure messages are commited (my code is not doing that)
- Ensure all channels are closed when the connection closes.

Whit [this change](https://github.com/oscarfh/micronaut-rabbitmq/commit/70d7aa50e23f400136d08af6f7a1222ebc6cf40b) I was able to create multiple consumers and receive the messages. The ack is still missing, but this was the POC I could create in 20 minutes. 

If my proposal from above make sense, I can create a PR. Let me know.
